### PR TITLE
ADAPT-2101: Replace CLSX with CNBuilder

### PR DESCRIPTION
**Pull request by jdwjdwjdw**
# READY FOR REVIEW

# Summary
- Replacing clsx-dedupe and clsx with [cnbuilder](https://www.npmjs.com/package/cnbuilder)

# Needed By (Date)
- When does this need to be merged by?

# Urgency
- How critical is this PR?

# Steps to Test

1. Do this
1. Then this
2. Then this

# Affected Projects or Products
- Does this PR impact any particular projects, products, or modules?

# Associated Issues and/or People
- JIRA ticket
- Other PRs
- Any other contextual information that might be helpful (e.g., description of a bug that this PR fixes, new functionality that it adds, etc.)
- Anyone who should be notified? (`@mention` them here)

# See Also
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)

### CODE REVIEW

**Comment by yvonnetangsu** [comment]
Yay - welcome @jdwjdwjdw 🎉  Fun user name 😂

**Comment by sherakama** [comment]
Looks like off to a good start. Compiles and runs.

**Comment by jdwjdwjdw** [comment]
> Yay - welcome @jdwjdwjdw 🎉 Fun user name 😂

Haha I've been meaning to update that, as well as my profile picture! Will get to it one of these days... 😂 And thank you! Really excited to contribute. 🎉

**Comment by jdwjdwjdw** [comment]
> Looks like off to a good start. Compiles and runs.

Yeah all of the components seemed to be performing as expected during my tests - there did seem to be some issues with the skiplink, which I will need to look into more. And will I need to add this as a PeerDependency, similar to clsx and clsx-dedupe?

**Comment by yvonnetangsu** [comment]
> > Looks like off to a good start. Compiles and runs.
> 
> Yeah all of the components seemed to be performing as expected during my tests - there did seem to be some issues with the skiplink, which I will need to look into more. And will I need to add this as a PeerDependency, similar to clsx and clsx-dedupe?

I'll let Shea answer the dependency question. I suspect the skiplink styles not showing up isn't cause by what you've done. I had similar issues last week - apparently it's a caching thing in the Netlify preview build. Shea told me to log into Netlify, and have it clear cache and redeploy. That fixed the issue for me.

**Comment by yvonnetangsu** [comment]
Different site, but this button here
Screen Shot 2021-03-26 at 5 34 10 PM

**Comment by sherakama** [comment]
Yeah, 

When we are working with branches, netlify can cache the branch and not the commit hash, so you may not see your changes. It is a good idea to clear cache on deploy in this case.

For the dependency issue, I think we can be safe in just placing it in the dependencies list. This a runtime dependency and shouldn't be a peer dependency.

**Comment by jdwjdwjdw** [comment]
Merged Yvonne's recent changes, then updated all additional files that needed clsx/clsxd replaced.
Cleared cache and redeployed, thanks for that tip! All components appeared to be working as expected from my review.
Should be ready for additional review.

**Review by sherakama — APPROVED** [review]
This looks good to me. @yvonnetangsu please review.

File: src/Button/Button.js [hunk 1]
```diff
@@ -66,15 +66,15 @@ export const Button = ({ className, children, onClick, variant, size, type, icon
   // Is disabled
   if (isDisabled) {
     levers.disabled = 'su-bg-black-20 su-text-black su-border-2 su-border-black-20 su-border-solid su-pointer-events-none';
-    levers.variant = clsxd(levers.variant, {'su-bg-digital-red': false, 'su-bg-white': false, 'su-text-digital-red': false, 'su-border-digital-red': false, 'hover:su-border-black': false, 'focus:su-border-black': false, 'su-text-white': false});
```
Comment (yvonnetangsu) [inline src/Button/Button.js:69]: @jdwjdwjdw Thanks for this PR - looks great. Just one question - is there a reason to remove the `'su-bg-white' : false` here?

Comment (jdwjdwjdw) [reply src/Button/Button.js:69]: @yvonnetangsu great catch! Looks like I missed those additions while doing my merge. Updating now...

Comment (yvonnetangsu) [reply src/Button/Button.js:69]: Awesome - thanks, if you could please do a final `npm run package` then it's GTG!

**Comment by yvonnetangsu** [comment]
GTG! Thanks @jdwjdwjdw 🎉

**Pull request merged**